lib/DiscoveryStreamFeed.jsm

```diff
@@ -145,28 +145,48 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
     }
   }
 
+  setupFeedPromise(url, newFeeds) {
+    const feedPromise = this.getComponentFeed(url);
+
+    feedPromise.then(data => {
+      newFeeds[url] = data;
```

Comment:
This is needed because the format we expected the finished data to be in wasn't what the promise.all would return, so we map it here as it finishes.

Reply:
Makes sense.